esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -6187,7 +6110,28 @@ bool CWsDfuEx::onDFUFileAccess(IEspContext &context, IEspDFUFileAccessRequest &r
     return true;
 }
 
-IGroup *CWsDfuEx::getDFUFileIGroup(const char *clusterName, ClusterType clusterType, const char *clusterTypeEx, StringArray &locations, StringBuffer &groupName)
+bool CWsDfuEx::onDFULookupFile(IEspContext &context, IEspDFULookupFileRequest &req, IEspDFUFileAccessResponse &resp)
+{
+    try
+    {
+        context.ensureFeatureAccess(FEATURE_URL, SecAccess_Read, ECLWATCH_DFU_ACCESS_DENIED, "WsDfu::DFULookupFile: Permission denied.");
```

Comment:
FYI the annotation in the method definition ""DfuAccess:READ":
ESPmethod [auth_feature("DfuAccess:READ")] DFULookupFile(DFULookupFileRequest, DFUFileAccessResponse);
generates this same context.ensureFeatureAccess call lower in the stack. We can either change the method declaration to:
auth_feature("DEFERRED"), //This declares that the method logic handles feature level authorization
or remove this block from the CWsDfuEx::onXYZ methods